benchmark.go

```diff
@@ -0,0 +1,435 @@
+			Bench:            getFlushBenchmark(1e5),
+			MinRuntime:       time.Second,
+			MaxRuntime:       10 * time.Minute,
+			Timeout:          20 * time.Minute,
+			IterationTimeout: 10 * time.Minute,
+			Count:            10,
+			MinIterations:    5,
+			MaxIterations:    20,
+			Recorder:         poplar.RecorderPerf,
+		},
+		{
+
+			CaseName:         "1MBBuffer",
+			Bench:            getFlushBenchmark(1e6),
+			MinRuntime:       time.Second,
+			MaxRuntime:       10 * time.Minute,
+			Timeout:          20 * time.Minute,
+			IterationTimeout: 10 * time.Minute,
+			Count:            10,
+			MinIterations:    5,
+			MaxIterations:    20,
+			Recorder:         poplar.RecorderPerf,
+		},
+		{
+
+			CaseName:         "10MBBuffer",
+			Bench:            getFlushBenchmark(1e7),
+			MinRuntime:       time.Second,
+			MaxRuntime:       10 * time.Minute,
+			Timeout:          20 * time.Minute,
+			IterationTimeout: 10 * time.Minute,
+			Count:            10,
+			MinIterations:    5,
+			MaxIterations:    20,
+			Recorder:         poplar.RecorderPerf,
+		},
+		{
+
+			CaseName:         "100MBBuffer",
+			Bench:            getFlushBenchmark(1e8),
+			MinRuntime:       time.Second,
+			MaxRuntime:       10 * time.Minute,
+			Timeout:          20 * time.Minute,
+			IterationTimeout: 10 * time.Minute,
+			Count:            10,
+			MinIterations:    5,
+			MaxIterations:    20,
+			Recorder:         poplar.RecorderPerf,
+		},
+	}
+}
+
+func getFlushBenchmark(bufferSize int) poplar.Benchmark {
+	numLines := bufferSize / lineLength
+	if numLines == 0 {
+		numLines = 1
+	}
+	buffer := make([]*internal.LogLine, numLines)
+	for i := 0; i < numLines; i++ {
+		ts := time.Now()
+		buffer[i] = &internal.LogLine{
+			Timestamp: &timestamp.Timestamp{Seconds: ts.Unix(), Nanos: int32(ts.Nanosecond())},
+			Data:      newRandCharSetString(lineLength),
+		}
+	}
+
+	return func(ctx context.Context, r poplar.Recorder, count int) error {
+		opts := &LoggerOptions{
+			Project:    "poplar-benchmarking",
+			TaskID:     newUUID(),
+			RPCAddress: rpcAddress,
+			Insecure:   true,
+		}
+
+		logger, err := MakeLogger(ctx, "benchmark", opts)
+		if err != nil {
+			return errors.Wrap(err, "problem creating buildlogger sender")
+		}
+
+		r.SetState(0)
+		b := logger.(*buildlogger)
+		for i := 0; i < count; i++ {
+			b.buffer = buffer
+			startAt := time.Now()
+			r.Begin()
+			if err := b.flush(); err != nil {
+				return errors.Wrap(err, "problem flushing")
+			}
+			r.End(time.Since(startAt))
+			r.IncOps(1)
+		}
+
+		r.SetState(1)
+		startAt := time.Now()
+		r.Begin()
+		if err = logger.Close(); err != nil {
+			return errors.Wrap(err, "problem closing buildlogger sender")
+		}
+		r.End(time.Since(startAt))
+		r.IncOps(1)
+
+		return nil
+	}
+
+}
+
+func newUUID() string {
+	b := make([]byte, 16)
+	_, _ = rand.Read(b)
+	return hex.EncodeToString(b)
+}
+
+var seededRand *rand.Rand = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+func newRandCharSetString(length int) string {
```

Comment:
should we move this to a util package

Reply:
we can, thought I do not know if it is worth it? it is only two small helper functions